src/customFunctions/customFetch.ts

```diff
@@ -0,0 +1,26 @@
+import { bearerTokenAuthorization } from "./../configs/bearerTokenAuthorization";
+
+export const customFetch = async (
+  input: RequestInfo,
+  init: RequestInit = {}
+) => {
+  init.headers = init.headers || bearerTokenAuthorization.headers;
+  const response = await fetch(input, init);
+
+  if (!response.ok) {
+    throw new CustomFetchError("Some problem with response: ", response);
+  }
```

Comment:
fun fact:
můžeš to zkrátit na one liner 

```js
if (!response.ok) throw new CustomFetchError("Some problem with response: ", response);
```

ale tohle je reálně samozřejmě úplně jedno